lib/routes/2048/index.ts

```diff
@@ -67,31 +67,32 @@ async function handler(ctx) {
     const id = ctx.req.param('id') ?? '3';
 
     const rootUrl = 'https://hjd2048.com';
-
+    // 获取地址发布页指向的 URL
     const domainInfo = (await cache.tryGet('2048:domainInfo', async () => {
-        const response = await ofetch('https://hjd.tw', {
+        const response = await ofetch('https://2048.info', {
             headers: {
                 accept: '*/*',
             },
         });
-        let $ = load(response);
-        const targetLink = new URL($('.address-list a').eq(0).attr('href'), 'https://hjd.tw').href;
-        const redirectResponse = await ofetch.raw(targetLink);
-        $ = load(redirectResponse._data);
-        return {
-            url: redirectResponse.url,
-            cookie:
-                $('script')
-                    .text()
-                    .match(/var safeid='(.*?)',/)?.[1] ?? '',
-        };
-    })) as { url: string; cookie: string };
-
-    const currentUrl = `${domainInfo.url}thread.php?fid-${id}.html`;
+        const $ = load(response);
+        const onclickValue = $('.button').first().attr('onclick');
+        const targetUrl = onclickValue.match(/window\.open\('([^']+)'/)[1];
+
+        return {url: targetUrl};
+    })) as { url: string};
+    // 获取重定向后的url和safeid
+    const redirectResponse = await ofetch.raw(domainInfo.url, {
+            headers: {
+                accept: '*/*',
+            },
```

Comment:
Unnecessary header. Requests from RSSHub will this added automatically 
https://github.com/DIYgod/RSSHub/blob/20ee222c3ffbc04404d762eada53adfdd6075cff/lib/utils/request-rewriter/fetch.ts#L41